src/controller/reducers/topicDashboard.js

```diff
@@ -0,0 +1,68 @@
+import {
+  RESOURCE_LOADING,
+  RESOURCE_SUCCESS,
+  ADD_RESOURCE,
+  UPDATE_RESOURCE,
+  DELETE_RESOURCE,
+  RESOURCE_ERROR
+} from '../actions/topicDashboard';
+
+import produce from 'immer';
+
+export const initialState = {
```

Comment:
Not sure this state layout is optimal. I think you should keep the topic object you receive from `RESOURCE_SUCCESS` on its own instead of arbitrarily factoring things like topicID and topicTitle out. For example, right now you don't have access to resourceOrder.

```suggestion
export const initialState = {
  resources: [],
  loading: false,
  error: null,
  topic: null // <- this will be the topic object populated by RESOURCE_SUCCESS
};
```